src/ray/gcs/subscription_executor.cc

```diff
@@ -0,0 +1,133 @@
+#include "ray/gcs/subscription_executor.h"
+
+namespace ray {
+
+namespace gcs {
+
+template <typename ID, typename Data, typename Table>
+Status SubscriptionExecutor<ID, Data, Table>::AsyncSubscribe(
+    const ClientID &client_id, const SubscribeCallback<ID, Data> &subscribe,
+    const StatusCallback &done) {
+  // TODO(micafan) Optimize the lock when necessary.
+  // Consider avoiding locking in single-threaded processes.
+  std::lock_guard<std::mutex> lock(mutex_);
+
+  if (subscribe != nullptr && subscribe_all_callback_ != nullptr) {
+    RAY_LOG(INFO) << "Duplicate subscription!";
```

Comment:
nit, this log isn't needed as we already return an invalid status.

Reply:
I think those log is useful in debug mode, so I will change log level to `DEBUG`.